PROTO_tests/tests/dtypes_test.py

```diff
@@ -0,0 +1,156 @@
+            dtypes.check_np_dtype("bigint")
+
+    def test_translate_np_dtype(self):
+        for b in [np.bool_, bool]:
+            assert ("bool", 1) == dtypes.translate_np_dtype(np.dtype(b))
+
+        for s in [np.str_, str]:
+            assert ("str", 0) == dtypes.translate_np_dtype(np.dtype(s))
+
+        assert ("int", 8) == dtypes.translate_np_dtype(np.dtype(np.int64))
+        assert ("uint", 8) == dtypes.translate_np_dtype(np.dtype(np.uint64))
+        assert ("float", 8) == dtypes.translate_np_dtype(np.dtype(np.float64))
+        assert ("uint", 1) == dtypes.translate_np_dtype(np.dtype(np.uint8))
+
+    def test_resolve_scalar_dtype(self):
+        assert "bool" == dtypes.resolve_scalar_dtype(True)
+        assert "int64" == dtypes.resolve_scalar_dtype(1)
+        assert "int64" == dtypes.resolve_scalar_dtype(-1)
+        assert "int64" == dtypes.resolve_scalar_dtype(np.int64(1))
+        assert "float64" == dtypes.resolve_scalar_dtype(float(0.0))
+        assert "float64" == dtypes.resolve_scalar_dtype(float(-1.0))
+        assert "float64" == dtypes.resolve_scalar_dtype(np.nan)
+        assert "float64" == dtypes.resolve_scalar_dtype(np.inf)
+        assert "float64" == dtypes.resolve_scalar_dtype(-np.inf)
+        assert "str" == dtypes.resolve_scalar_dtype("test")
+        assert "<class 'list'>" == dtypes.resolve_scalar_dtype([1])
+        assert "uint64" == dtypes.resolve_scalar_dtype(2**63 + 1)
+        assert "bigint" == dtypes.resolve_scalar_dtype(2**64)
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_pdarrays_datatypes(self, size):
+        assert dtypes.dtype("int64") == ak.array(np.arange(size)).dtype
+        assert dtypes.dtype("uint64") == ak.array(np.arange(size), ak.uint64).dtype
+        assert dtypes.dtype("bool") == ak.ones(size, ak.bool).dtype
+        assert dtypes.dtype("float64") == ak.ones(size).dtype
+        assert dtypes.dtype("str") == ak.array([f"string {i}" for i in range(size)]).dtype
+        assert (
+            dtypes.dtype("bigint")
+            == ak.bigint_from_uint_arrays(
+                [ak.ones(size, dtype=ak.uint64), ak.arange(size, dtype=ak.uint64)]
+            ).dtype
+        )
+        assert dtypes.dtype("bigint") == ak.array([i for i in range(2**200, 2**200 + size)]).dtype
+
+    @pytest.mark.parametrize("supported", [1, np.int64(1), np.int64(1.0), 1, np.uint32(1)])
+    @pytest.mark.parametrize("unsupported", [1.0, "1"])
+    def test_isSupportedInt(self, supported, unsupported):
+        assert dtypes.isSupportedInt(supported)
+        assert not dtypes.isSupportedInt(unsupported)
+
+    @pytest.mark.parametrize("supported", [1.0, float(1), np.float64(1), np.float64(1.0)])
+    @pytest.mark.parametrize("unsupported", [np.int64(1.0), int(1.0), "1.0"])
+    def test_isSupportedFloat(self, supported, unsupported):
+        assert dtypes.isSupportedFloat(supported)
+        assert not dtypes.isSupportedFloat(unsupported)
```

Comment:
Same deal here we are checking the same values are supported or not over and over

Reply:
Updated